src/PatchOrchestrationApplication/NodeAgentSFUtility/src/Helpers/HealthManagerHelper.cs

```diff
@@ -49,23 +54,92 @@ class HealthManagerHelper
 
             try
             {
-                ServiceHealthReport healthReport = new ServiceHealthReport(new Uri(applicationName + ServiceNameSuffix), healthInformation);
-                fabricClient.HealthManager.ReportHealth(healthReport);
+                ServiceHealthReport healthReport = new ServiceHealthReport(new Uri(applicationName + serviceNameSuffix), healthInformation);
+                Func<string, bool> condition = (s) => {
+                    bool serviceExists = CheckIfServiceIsUp(fabricClient, applicationName, s).GetAwaiter().GetResult();
+                    if (serviceExists)
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                };
+                HealthReportSendOptions sendOptions = new HealthReportSendOptions();
+                sendOptions.Immediate = true;
+                TimeoutProcessWithRetry(() =>
+                {
+                    fabricClient.HealthManager.ReportHealth(healthReport, sendOptions);
+                }, condition,
+                serviceNameSuffix, timeout
+                );
+
+                Task.Delay(TimeSpan.FromSeconds(2)).GetAwaiter().GetResult();
```

Comment:
Why this logic and timeout?

Reply:
timeout is 120 sec which is allotted for the operation to complete. this logic was kept because when we deploy POA sometimes, events of some of the nodes are missed because till that time Coordinator Service is not up